sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -76,6 +76,22 @@ private static AmqpMessage BuildAmqpBatchFromMessage(IReadOnlyCollection<Service
             return BuildAmqpBatchFromMessages(batchMessages, firstMessage, forceBatch);
         }
 
+        /// <summary>
+        ///   Builds a batch of <see cref="AmqpMessage" /> from a set of <see cref="ServiceBusMessage" />
+        ///   optionally propagating the custom properties.
+        /// </summary>
+        ///
+        /// <param name="source">The set of messages to use as the body of the batch message.</param>
+        /// <param name="firstMessage">The first message being sent in the batch.</param>
+        /// <param name="forceBatch">Set to true to force creating as a batch even when only one message.</param>
+        ///
+        /// <returns>The batch <see cref="AmqpMessage" /> containing the source messages.</returns>
+        ///
+        public virtual AmqpMessage BuildAmqpBatchFromMessages(IReadOnlyCollection<AmqpMessage> source, ServiceBusMessage firstMessage, bool forceBatch)
```

Comment:
As above, not sure how to work this without requiring the `ServiceBusMessage` first message.

Reply:
I think we'll need @JoshLove-msft's thought... but since we have a collection, couldn't we just grab the message at index 0?

Reply:
I wasn't sure how to get the right properties off the Amqp message, maybe convert it back to a ServiceBusMessage? Initially, I just put all the converted messages in an AMQP batch and didn't populate the batch envelope properties (see L133) because I couldn't get them from the AmqpMessage but I was getting an error (like [this](https://github.com/Azure/azure-amqp/issues/118#issuecomment-418546828)) that I fixed by adding the messageId back in using the first message.

Right now we just have AmqpMessage -> ReceivedServiceBusMessage, just add an AmqpMessage -> ServiceBusMessage?